tests/ui/for_loop.stderr

```diff
@@ -506,5 +506,73 @@ help: use the corresponding method
 344 |     for k in rm.keys() {
     |         ^
 
-error: aborting due to 51 previous errors
+error: it looks like you're manually copying between slices
+   --> $DIR/for_loop.rs:393:5
+    |
+393 | /     for i in 0..src.len() {
+394 | |         dst[i] = src[i];
+395 | |     }
+    | |_____^ help: try replacing the loop by: `dst.clone_from_slice(&src[..])`
+    |
+    = note: `-D manual-memcpy` implied by `-D warnings`
+
+error: it looks like you're manually copying between slices
+   --> $DIR/for_loop.rs:398:5
+    |
+398 | /     for i in 0..src.len() {
+399 | |         dst[i + 10] = src[i];
+400 | |     }
+    | |_____^ help: try replacing the loop by: `dst[10..].clone_from_slice(&src[..])`
+
+error: it looks like you're manually copying between slices
+   --> $DIR/for_loop.rs:403:5
+    |
+403 | /     for i in 0..src.len() {
+404 | |         dst[i] = src[i + 10];
+405 | |     }
+    | |_____^ help: try replacing the loop by: `dst.clone_from_slice(&src[10..])`
+
+error: it looks like you're manually copying between slices
+   --> $DIR/for_loop.rs:408:5
+    |
+408 | /     for i in 11..src.len() {
+409 | |         dst[i] = src[i - 10];
+410 | |     }
+    | |_____^ help: try replacing the loop by: `dst[11..].clone_from_slice(&src[(11 + -10)..(src.len() + -10)])`
```

Comment:
I think we should not produce output like `a + -b`, can you try to detect this situation, as it seems to be rather common.